src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -263,7 +263,14 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
 
             if (attributeConstructor is not null)
             {
-                CheckRequiredMembersInObjectInitializer(attributeConstructor, ImmutableArray<BoundExpression>.CastUp(boundNamedArguments), node, diagnostics);
+                if (IsEarlyAttributeBinder)
+                {
+                    diagnostics.Add(new LazyRequiredMembersDiagnosticsInfo(attributeConstructor, ImmutableArray<BoundExpression>.CastUp(boundNamedArguments), node), node.GetLocation());
```

Comment:
>diagnostics.Add(new LazyRequiredMembersDiagnosticsInfo(attributeConstructor, ImmutableArray<BoundExpression>.CastUp(boundNamedArguments), node), node.GetLocation()); [](http://example.com/codeflow?start=20&length=165)

A comment on ```EarlyWellKnownAttributeBinder``` says: "It only binds those attribute argument syntax which can produce valid attribute arguments, but doesn't report any diagnostics. Subsequent binding phase will rebind such erroneous attributes and generate appropriate diagnostics."

Either the comment is not accurate, or it feels like we don't need to have any code in this branch. Especially given that the interesting part of  ```LazyRequiredMembersDiagnosticsInfo```'s implementation is unreachable.

Reply:
Looking deeper, it looks like what we are doing here will cause ```out bool generatedDiagnostics``` to be true for all ```EarlyWellKnownAttributeBinder.GetAttribute``` calls. That value affects how result of the calls is processed in a fundamental way. It looks like by doing this, we are having an unconditional effect on all those callers without even knowing that we have anything to report.

Reply:
At the moment I see the following alternatives:
1. Do what we do for ```IsAttributeConditionallyOmitted``` helper. I.e. completely skip required members analysis during early attribute binding and then explicitly do it for the early bound attributes later, when we bind remaining attributes in a regular mode.
2. Move the check to ```LoadAndValidateAttributes``` to the section that does some post processing of attributes, including nullable analysis.

I am personally leaning towards the second option. In my opinion, the required members check isn't supposed to invalidate an attribute application. Yes, language wants to report an additional diagnostics for an attribute in some cases, but the attribute application should still be considered good if overload resolution succeeded and other runtime requirements are satisfied.